### PR TITLE
Backend/bugtool/1013 friend accept notification email

### PR DESCRIPTION
**Pull request by austinburnett**
**Backend checklist**
- [ ] Formatted my code by running `isort . && black .` in `app/backend`
- [ ] Added tests for any new code or added a regression test if fixing a bug
- [ ] All tests pass
- [ ] Run the backend locally and it works
- [ ] Added migrations if there are any database changes, rebased onto `develop` if necessary for linear migration history

**Frontend checklist**
- [ ] Formatted my code with `yarn format && yarn lint --fix`
- [ ] There are no warnings from `yarn lint`
- [ ] There are no console warnings when running the app
- [ ] Added any new components to storybook
- [ ] Added tests where relevant
- [ ] All tests pass
- [ ] Clicked around my changes running locally and it works
- [ ] Checked Desktop, Mobile and Tablet screen sizes

### CODE REVIEW

File: app/backend/src/couchers/servicers/api.py [hunk 1]
```diff
@@ -644,6 +644,9 @@ def RespondFriendRequest(self, request, context):
             friend_request.status = FriendStatus.accepted if request.accept else FriendStatus.rejected
             friend_request.time_responded = func.now()
 
+            if friend_request.status == FriendStatus.accepted:
+                send_friend_request_accepted_email(FriendRelationship)
```
Comment (aapeliv) [inline app/backend/src/couchers/servicers/api.py]: `FriendRelationship` is the class name; you want to refer to the instance itself, `friend_request`

File: app/backend/src/couchers/tasks.py [hunk 2]
```diff
@@ -140,6 +140,20 @@ def send_friend_request_email(friend_relationship):
         },
     )
 
+def send_friend_request_accepted_email(friend_relationship):
+    friend_requests_link = urls
```
Comment (aapeliv) [inline app/backend/src/couchers/tasks.py]: This should be a link to the person's profile who received the original friend req.

It should be `urls.profile_link(friend_relationship.to_user.username)` or something. You might need to add a `profile_link` function into `urls.py`. The right URL is in the frontend somewhere (see the comment at the top of `urls.py`)

**Review by aapeliv — CHANGES_REQUESTED** [review]
Great start! See comments

**Comment by austinburnett** [comment]
@aapeliv If you want you can just wait a couple more days to review this pr. I'm getting ready to make some more changes.

**Comment by aapeliv** [comment]
I'm not sure what you've done, but this is now picking up a whole lot of unrelated commits and diff contains too much stuff not related to this PR. Can you message me on Slack so we can sort it out?

**Comment by aapeliv** [comment]
Test failed: https://gitlab.com/couchers/couchers/-/jobs/1389893220#L153

File: app/backend/templates/friend_request_accepted.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+---
+subject: "{{ friend_relationship.to_user.name|couchers_escape }} accpeted your friend request!"
```
Comment (aapeliv) [inline app/backend/templates/friend_request_accepted.md]: Typo in "accepted"

File: app/backend/src/couchers/urls.py [hunk 4]
```diff
@@ -14,6 +14,10 @@ def profile_link():
     return f"{config['BASE_URL']}/profile"
 
 
+def user_link(user_name):
+    return f"{config['BASE_URL']}/user/{user_name}"
```
Comment (aapeliv) [inline app/backend/src/couchers/urls.py]: Please rename `user_name` with `username`

File: app/backend/src/tests/test_email.py [hunk 5]
```diff
@@ -197,6 +198,42 @@ def test_friend_request_email(db):
...
+        assert "friend" in subject.lower()
+        assert from_user.name in plain
+        assert from_user.name in html
+        assert to_user.name in subject
+        assert to_user.name in plain
+        assert to_user.name in html
+        assert to_user.avatar.thumbnail_url not in plain                 
+        assert to_user.avatar.thumbnail_url in html                         
+        #assert f"{config['BASE_URL']}/to_user.name" in plain
+        #assert f"{config['BASE_URL']}/to_user.name" in html
```
Comment (aapeliv) [inline app/backend/src/tests/test_email.py]: `assert f"{config['BASE_URL']}/user/{to_user.username}" in plain`

File: app/backend/src/couchers/tasks.py [hunk 6]
```diff
@@ -176,6 +176,22 @@ def send_friend_request_email(friend_relationship):
     )
 
 
+def send_friend_request_accepted_email(friend_relationship):
+    friend_requests_link = urls.user_link(friend_relationship.to_user.username)
```
Comment (aapeliv) [inline app/backend/src/couchers/tasks.py]: Should be renamed to `user_link`

**Review by aapeliv — CHANGES_REQUESTED** [review]
Few tiny things left

**Comment by aapeliv** [comment]
Tests are still failing: https://gitlab.com/couchers/couchers/-/jobs/1439757892

**Review by aapeliv — APPROVED** [review]
LGTM!

**Pull request merged**